### ISSUE TITLE
GeoShape failure with some polygons

### ISSUE THREAD

**Issue by gaellafond** [elastic/elasticsearch#89059]
### Elasticsearch Version

8.X (replicated with 8.1.2 and 8.3.3)

### Installed Plugins

None

### Java Version

OpenJDK 11.0.15

### OS Version

Ubuntu 20.04.4 LTS

### Problem Description

We are running Elastic Search in a Docker container:
https://hub.docker.com/_/elasticsearch

Indexation is done using a Java application we are currently developing.

We created an index using the following settings:

```
.properties("wkt", new Property.Builder()
    .geoShape(new GeoShapeProperty.Builder()
        .ignoreZValue(true)
        .coerce(true) // Automatically close polygons
        .build())
    .build())
```

It works with about 99% of the GeoShape (WKT) we are indexing. But the following polygon trigger an error for some reason:

```
POLYGON ((-175.4736328125 -31.245117187500004, -171.2548828125 36.9580078125, 171.8701171875 39.0673828125, 175.3857421875 -30.5419921875, -32.0361328125 -34.0576171875, -175.4736328125 -31.245117187500004))
```

Simplified polygon which still show the issue:
```
POLYGON ((-91 9, 0 80, 90 10, -90 9, -91 9))
```

Error:
```
co.elastic.clients.elasticsearch._types.ElasticsearchException - [es/index] failed: [mapper_parsing_exception] failed to parse field [wkt] of type [geo_shape]
```

We have the same issue when indexing the land from the following Natural Earth Data:
https://www.naturalearthdata.com/downloads/50m-physical-vectors/

We workaround the issue by swapping **GeoShape** with **Shape**, but I suspect we are losing some useful features.

### Steps to Reproduce

1. Install Elastic Search 8.3.3
2. Create an index with a geo_shape field
3. Index the following polygon: `POLYGON ((-91 9, 0 80, 90 10, -90 9, -91 9))`
4. Elastic Search throw a `ElasticsearchException`

**Comment by craigtaverner** [comment]
This polygon is both clockwise (non-standard) and very wide (contains edges longer than 180 degrees). Elasticsearch appears to automatically handle clockwise polygons only if they do not contain such edges. If all the polygons in the dataset are clockwise, you can set that in the mapping using the `orientation:clockwise` setting in the mapping, as described here: https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-shape.html#geo-shape-mapping-options

If only this polygon is clockwise, it is better to fix the polygon to the standard counterclockwise orientation by re-ordering the points (reversing the order).

**Comment by gaellafond** [comment]
@craigtaverner Thank you! All our polygons are indeed clockwise. I'm not sure why most (all?) software we are using creates clockwise polygons. Why is it so hard to follow the standards?

The default in ElasticSearch is "Right", which I assumed meant clockwise. As it turns out, right is counter-clockwise and left is clockwise.

I changed the field setting to "left" and that fixed the issue.
```
.properties("wkt", new Property.Builder()
    .geoShape(new GeoShapeProperty.Builder()
        .ignoreZValue(true)
        .coerce(true) // Automatically close polygons
        .orientation(GeoOrientation.Left) // Default: right (counter-clockwise)
        .build())
    .build())
```

**Comment by craigtaverner** [comment]
Glad to have helped, @gaellafond.

If you're interested in how `right` maps to counter-clockwise, imagine holding your right hand above the map, curling your fingers in a counter-clockwise direction. Your thumb then sticks upwards, out of the map, towards your face (positive z direction). If you do the same with your left hand, positive z happens only with a clockwise orientation of the fingers. This `right hand rule` is similar to one used in electromagnetism (relationship between electric and magnetic fields).

**Comment by gaellafond** [comment]
Thank you for the right / left explanation. It makes sense.

I was not feeling comfortable using `clockwise` for polygons in the index, so I set it back to `counter-clockwise` and used the JTS library to fix the orientation of polygons.

The JTS library have a `isCCW` method and a `reverse` method, which can be used to make sure all polygons are standard in the index, no mater what state they comes in.

**Comment by craigtaverner** [comment]
Cool, @gaellafond! I've used JTS myself for various things, but not that it turns out. Nice find!
